bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1902,6 +1950,15 @@ private static boolean checkIsServiceType(Type sourceType, Type targetType, List
         return false;
     }
 
+    private static boolean checkIsClientType(Type sourceType, Type targetType, List<TypePair> unresolvedTypes) {
+        if (sourceType.getTag() == TypeTags.CLIENT_TAG) {
+            return checkObjectEquivalency(sourceType, (BObjectType) targetType, unresolvedTypes);
+        }
+
+        // TODO: Check for client in the flags
```

Comment:
Do we need this? Shall we directly  return sourceType.getTag() == TypeTags.CLIENT_TAG && checkObjectEquivalency(sourceType, (BObjectType) targetType, unresolvedTypes)

Reply:
`checkIsServiceType` uses this approach. Not sure we need this for `checkIsClientType`

https://github.com/ballerina-platform/ballerina-lang/blob/a1af3b4c1c72ca713a827fb2716346fff08b3116/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java#L1939-L1950